lib/rucio/transfertool/fts3.py

```diff
@@ -615,7 +615,7 @@ def _transfer_link(self) -> str:
 
     def _find_attribute_updates(self, request: dict, new_state: RequestState, reason: str, overwrite_corrupted_files: Optional[bool] = None) -> Optional[dict[str, Any]]:
         attributes = None
-        if new_state == RequestState.FAILED and 'Destination file exists and overwrite is not enabled' in (reason or ''):
+        if new_state == RequestState.FAILED and ('Destination file exists and overwrite is not enabled' in (reason or '') or 'Destination file exists and is on tape' in (reason or '')):
```

Comment:
I think it would also make sense to clarify that these error messages cannot both happen in the same site: `'Destination file exists and overwrite is not enabled'` is for FTS < 3.12.12 and `'Destination file exists and is on tape'` is for FTS >= 3.12.12, but without this context, one might think that these are separate errors that can both occur within the same FTS instance on a given site.

I think it could be a good idea to have these strings be variables somewhere in the file, e.g.:

```
fts_file_exists_error_msg = ''Destination file exists and is on tape'
fts_file_exists_error_msg_legacy = 'Destination file exists and overwrite is not enabled'  # Error message used in FTS < 3.12.12, checked in Rucio for backwards compatibility
```

And then accessing them in `_find_attribute_updates` and `_is_recoverable_fts_overwrite_error`. This way, we also avoid the duplication of writing out these strings both times each